### PR TITLE
Woo POS: Simple products info dialog

### LINKED ISSUE #12100: [Woo POS] M2: Display Simple products only information dialog
The goal of this task is to add an icon to the products toolbar that serves as an entry point for an information dialog, educating users on why the POS displays only Simple products.

Screenshot 2024-07-22 at 3 05 08 PM
Screenshot 2024-07-22 at 3 05 17 PM

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #12100 

### Description

This PR introduces a dialog for displaying more information about why some products are not shown within the POS system. The new feature enhances user experience by providing detailed product information through a modal dialog when users interact with the info icon.

### Design
Screenshot 2024-07-29 at 4 14 40 PM

### Note to reviewer
"Create an order in store management" CTA has been removed and is out of scope for the POC

### Changes:

1. Extracted products heading into a toolbar.
2. Added logic and event handling for the product info icon click.
3. Implemented a new ProductInfoDialog for displaying product details.
4. Integrated the dialog into the home screen and product screen.
5. Refactored the view model to handle product info display logic.
6. Added comprehensive tests to verify the dialog's functionality and correctness.
7. Applied necessary UI changes and fixed detected issues.

### Testing information

There are 2 entry points for this dialog
1. By clicking "Learn more" link in Simple products only banner
2. By clicking the info icon from the products toolbar in the products screen - Note that this info icon will only be visible when the Simple products only banner is dismissed by the user

## Entry point 1
1. Navigate to the POS mode from app settings
2. Ensure you see Simple products only banner
3. Click on Learn more link
4. Ensure you see Products info dialog.

## Entry point 2
1. Navigate to the POS mode from app settings
2. Ensure you **do not** see Simple products only banner
3. Ensure you see an info icon in the products screen toolbar
4. Click on the icon
5. Ensure you see Products info dialog

### Images/gif

https://github.com/user-attachments/assets/80376192-4123-4ccb-8d5a-d47b5d7739f1

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeState.kt [hunk 1]
```diff
@@ -21,6 +23,21 @@ data class WooPosHomeState(
...
+    sealed class ProductsInfoDialog {
+        data object Hidden : ProductsInfoDialog()
+
+        data class Visible(
+            @StringRes val header: Int,
+            @StringRes val primaryMessage: Int,
+            @StringRes val secondaryMessage: Int,
+            val primaryButton: PrimaryButton,
+        ) : ProductsInfoDialog() {
+            data class PrimaryButton(
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeState.kt:36]: 💡  I'm curious what was the reason for wrapping the button label in a new data class. Probably, we are going to need only the button label in the state model.

**Review by samiuelson — APPROVED** [review]
LGTM 🚀 

I left a non-blocking question. Also, I can see the dialogs disappears when activity is recreated; probably not a big deal at the moment though.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeState.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/WooPosHomeState.kt:36]: Initially, I had defined the button click action within the state. However, I later realised that the POS module handles UI events differently, allowing me to use the onDismissRequest for both the primary button and close action instead.

**Comment by AnirudhBhat** [comment]
>Also, I can see the dialogs disappears when activity is recreated; probably not a big deal at the moment though.

Thanks for noticing this! I've fixed it https://github.com/woocommerce/woocommerce-android/pull/12162/commits/15f3503a92ed342077d79dbeac589436d00e54fa


https://github.com/user-attachments/assets/1823c3d6-fb6b-4625-9034-642749415b84

**Pull request merged**